kernel/pmu_x86/src/lib.rs

```diff
@@ -696,7 +696,7 @@ pub fn start_samples(event_type: EventType, event_per_sample: u32, task_id: Opti
     // and is therefore by definition in the range `u32::MIN..=u32::MAX`.
     // We'll check anyways, just in case `event_per_sample`'s type is changed.
     #[allow(clippy::absurd_extreme_comparisons)]
-    if event_per_sample > core::u32::MAX || event_per_sample < core::u32::MIN {
+    if !(core::u32::MIN..=core::u32::MAX).contains(&event_per_sample) {
```

Comment:
might be better to change this to something like `u32::try_from(event_per_sample)` rather than this complex conditional